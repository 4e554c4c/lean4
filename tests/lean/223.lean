universes u v

inductive Imf {α : Type u} {β : Type v} (f : α → β) : β → Type (max u v)
| mk : (a : α) → Imf f (f a)

def h {α β} {f : α → β} : {b : β} → Imf f b → α
| Imf.mk a => a

#print h

theorem ex : {α β : Sort u} → (h : α = β) → (a : α) → cast h a ≅ a
  | rfl, a => HEq.refl a

#print ex
