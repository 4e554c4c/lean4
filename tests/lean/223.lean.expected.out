def h.{u_1, u_2} : {α : Type u_1} → {β : Type u_2} → {f : α → β} → {b : β} → Imf f b → α :=
fun {α : Type u_1} {β : Type u_2} {f : α → β} (x : β) (x_1 : Imf f x) =>
  match x, x_1 with 
  | .(f a), Imf.mk a => a
theorem ex.{u} : ∀ {α β : Sort u} (h : α = β) (a : α), cast h a ≅ a :=
fun (x x_1 : Sort u) (x_2 : x = x_1) (x_3 : x) =>
  match x, x_1, x_2, x_3 with 
  | x, .(x), Eq.refl x, a => HEq.refl a
